expirationd.lua

```diff
@@ -327,48 +390,202 @@ end
---                                or when iterator is 'EQ', iterate over tuples with this index value;
---                                must be a value of the same data type as the index field or fields,
---                                or a function which returns such value;
---                                if omitted or nil, all tuples will be checked.
---     * tuples_per_iteration  -- Number of tuples to check in one batch (iteration); default is 1024.
---     * atomic_iteration      -- Boolean, false (default) to process each tuple as a single transaction;
---                                true to process tuples from each batch in a single transaction.
---     * process_while         -- Function to call before checking each tuple;
---                                if it returns false, the task will stop until next full scan.
---     * iterate_with          -- Function which returns an iterator object which provides tuples to check,
---                                considering the start_key, process_while and other options.
---                                There's a default function which can be overriden with this parameter.
---     * on_full_scan_start    -- Function to call before starting a full scan iteration.
---     * on_full_scan_complete -- Function to call after completing a full scan iteration.
---     * on_full_scan_success  -- Function to call after successfully completing a full scan iteration.
---     * on_full_scan_error    -- Function to call after terminating a full scan due to an error.
---     * args                  -- Passed to is_tuple_expired and
---                                process_expired_tuple() as additional context.
---     * full_scan_time        -- Time required for a full index scan (in seconds).
---     * iteration_delay       -- Max sleep time between iterations (in seconds).
---     * full_scan_delay       -- Sleep time between full scans (in seconds).
---     * force                 -- Run task even on replica.
---  }
+--- Run a scheduled task to check and process (expire) tuples in a given space.
+--
+-- How expirationd works in general:
+--
+-- 1. Process min(`space_length`, `tuples_per_iteration`) tuples at once.
+--
+-- 2. Sleep `tuples_per_iteration` × `full_scan_time` / `space_length` (but not
+--    beyond 1 second).
+--
+-- 3. Repeat 1-2 until the whole space will be traversed.
+--
+-- 4. Sleep 1 second.
+--
+-- 5. Repeat 1-4.
+--
+--
+-- @string name
+--     Task name.
+-- @string space_id
+--     Space to look in for expired tuples.
+-- @func is_tuple_expired
+--     Function, must accept tuple and return `true` or `false` (is tuple
+--     expired or not), receives `args` and `tuple` as arguments.
+--
+-- Example of function:
+--
+-- ```
+-- local function is_tuple_expired(args, tuple)
+--     local tuple_expire_time = get_field(tuple, args.field_no)
+--     if type(tuple_expire_time) ~= 'number' then
+--         return true
+--     end
```

Comment:
I don't spot the idea of this check. If there is no expire information within a tuple, expire it anyway?

Reply:
Fixed it. (copy-pasted from test)

```lua
 -- local function is_tuple_expired(args, tuple)
 --     local tuple_expire_time = get_field(tuple, args.field_no)
---     if type(tuple_expire_time) ~= 'number' then
---         return true
---     end
 --     local current_time = fiber.time()
---
 --     return current_time >= tuple_expire_time
 -- end
```